stock_orderpoint_origin/models/procurement_group.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2021 Open Source Integrators
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+
+
+class ProcurementGroup(models.Model):
+    _inherit = "procurement.group"
+
+    @api.model
+    def run(self, procurements, raise_user_error=True):
+        # Store the reorder source Procurement Groups
+        # To be used for reference and link navigation
+        # Also updates the Origin field of the PO
+        Forecast = self.env["report.stock.report_product_product_replenishment"]
```

Comment:
That sounds strange to add business data from reporting data ...

Reply:
How else can you know what originated the demand?